geometry/eolearn/geometry/transformations.py

```diff
@@ -134,15 +145,41 @@ def _get_vector_data(self, eopatch):
         feature_type, feature_name = next(self.vector_input(eopatch))
         return eopatch[feature_type][feature_name]
 
-    def _get_rasterization_shapes(self, eopatch, group_classes=False):
+    def _sort_timestamps(self, eopatch):
+        """ indexes all timestamps found in the vector input data according to the eopatch timestamps. Any 
+        timestamps present in the vector data but missing from the EOPatch are ignored. The exception to this
+        is if no timestamps are found in the EOPatch, they are added from the vector data. 
+        """
+        vectors = self._get_vector_data(eopatch)
+        vector_timestamps = set(vectors['TIMESTAMP'])
+
+        if len(eopatch.timestamp )==0:
+            eopatch.timestamp = list(vector_timestamps)
+            warnings.warn('The EOPatch has no timestamp data. Timestamps from the vector data will be added '
+                          'to the EOPatch', RuntimeWarning)
+
+        if any([t_vec not in eopatch.timestamp for t_vec in vector_timestamps]):
+            warnings.warn('Some timestamps in the vector data are not found in the eopatch timestamps and will be '
+                          'ignored', RuntimeWarning)
+
+        timestamp_flags = [(idx, time, 1) if time in vector_timestamps else (idx, time, 0) for idx, time in enumerate(eopatch.timestamp)]
```

Comment:
I would like to avoid undescriptive flags that have an effect in different part of the code. Instead, you can simply return a set of timestamps for which some data exists. Then in the execute method you can do something like:

```python
for timestamp in eopatch.timestamp:
    if timestamp not in timestamps_in_vector_data
        continue:
    ...
```

Reply:
Done